cli/clibase/option.go

```diff
@@ -79,6 +95,85 @@ func (o Option) YAMLPath() string {
 // OptionSet is a group of options that can be applied to a command.
 type OptionSet []Option
 
+// UnmarshalJSON implements json.Unmarshaler for OptionSets. Options have an
+// interface Value type that cannot handle unmarshalling because the types cannot
+// be inferred. Since it is a slice, instantiating the Options first does not
+// help.
+//
+// However, we typically do instantiate the slice to have the correct types.
+// So this unmarshaller will attempt to find the named option in the existing
+// set, if it cannot, the value is discarded. If the option exists, the value
+// is unmarshalled into the existing option, and replaces the existing option.
+//
+// The value is discarded if it's type cannot be inferred. This behavior just
+// feels "safer", although it should never happen if the correct option set
+// is passed in.
+func (os *OptionSet) UnmarshalJSON(data []byte) error {
```

Comment:
It would probably be good to test this by validating that we can marshal/unmarshal an option and get the same thing back (except if it's secret, in which case we should get an empty-valued option)

Reply:
The `server_test` does test the unmarshal, but I can write an additional test.

Reply:
Wrote the test :+1: